### ISSUE TITLE
India lists only 1 region, it appears to use an odd file AddRegionsForIndia.php

### ISSUE THREAD

**Issue by devchris79** [magento/magento2#39102]
### Preconditions and environment

- Magento version 2.4.7-p2

### Steps to reproduce

Select India, only 1 region appears - Ladakh

### Expected result

Multiple regions displayed

### Actual result

1 region appears - Ladakh

### Additional information

It appears that a selection of India is using `app/code/Magento/Directory/Setup/Patch/Data/AddRegionsForIndia.php` I assume it should be using `app/code/Magento/Directory/Setup/Patch/Data/AddDataForIndia.php` (file also exists in same directory) as every other country appears to use this format. India appears to be some odd exception...

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by devchris79** [comment]
@magento give me 2.4-develop instance

**Comment by devchris79** [comment]
The development instance doesn't show the issue, India has the regions in the shipping options.
I wonder if its an issue when updating an instance of Magento.

**Comment by engcom-Bravo** [comment]
Hi @devchris79,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is not reproducible.Multiple regions displayed




<img width="654" alt="Screenshot 2024-08-26 at 11 49 14" src="https://github.com/user-attachments/assets/8ad6a972-ef91-4db6-a068-f4a33d119550">


Could you please let us know from which version you have upgraded the instance and facing this issue.

Thanks.

**Comment by devchris79** [comment]
Magento was updated from version 2.4.6 to 2.4.7, although the issue could have been present prior to that. I have manually injected regions into the database to correct for now.

@glo17720 (file author) Why is there two very similar files for adding regions to India, [AddRegionsForIndia.php](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Directory/Setup/Patch/Data/AddRegionsForIndia.php) looks wrong with only Ladakh in the array as seen in our 2.4.7 site, where as [AddDataForIndia.php](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Directory/Setup/Patch/Data/AddDataForIndia.php) looks correct.

**Comment by engcom-Bravo** [comment]
Hi @devchris79,

Thanks for your update.

We are moving this issue `On hold` as we are discussing this issue with the Internal Team.

Thanks

**Comment by engcom-Bravo** [comment]
Hi @devchris79,

We got reply from the internal team as follows :

In Magento 2, data patches are applied during bin/magento setup:upgrade. When a new data patch is added, Magento checks the patch_list table to see if it’s already applied. If not, it applies the patch.

If you modify an existing patch, it won’t reapply automatically. To reapply, you must delete the patch’s entry in `patch_list` or create a new patch file for additional changes.

The file AddDataForIndia.php is a data patch file that is applied once during installation.

To reapply the data patch with the newly added Ladakh region in the AddDataForIndia.php file, follow these steps:

Remove the entry for the `AddDataForIndia` patch from the `patch_list` table.

```
   SET SQL_SAFE_UPDATES = 0;
   DELETE FROM <root-directory-name>.patch_list WHERE patch_name = 'Magento\\Directory\\Setup\\Patch\\Data\\AddDataForIndia';
```

Delete the corresponding regions associated with India (to avoid duplicate regions when the patch is reapplied) from the `directory_country_region` table.

   ```
DELETE FROM <root-directory-name>.directory_country_region WHERE country_id = 'IN';
   SET SQL_SAFE_UPDATES = 1;
```

Note: Being in production, deleting directly from SQL is not recommended.

This will force Magento to reapply the patch.

To avoid these steps, a new data patch, `AddRegionsForIndia`, has been added with the new region, Ladakh. This patch will be applied without affecting existing data.

Below are screen recordings of regions in versions 2.4.6 and 2.4.7, showing that regions are unaffected after upgrading from 2.4.6 to 2.4.7.



https://github.com/user-attachments/assets/9bd5abf5-0979-4dda-98d4-76499122fc00





https://github.com/user-attachments/assets/25c2ddc2-b442-49df-91ae-c0d3bce3cab8



verified it by upgrading from 2.4.6 to 2.4.7. Issue is not reproducible.Hence we are closing this issue.

Thank
